src/editors/data/redux/thunkActions/problem.js

```diff
@@ -0,0 +1,16 @@
+import _ from 'lodash-es';
+import { actions } from '..';
+import { OLXParser } from '../../../containers/ProblemEditor/data/OLXParser';
+import { parseSettings } from '../../../containers/ProblemEditor/data/SettingsParser';
+
+export const initializeProblem = (blockValue) => (dispatch) => {
+  const rawOLX = _.get(blockValue, 'data.data', {});
+  const olxParser = new OLXParser(rawOLX);
+  let { settings, ...data } = olxParser.getParsedOLXData();
```

Comment:
Can this be in a try... catch loop, as we might open olx which is not amenable to being viewed in the digital editor?